### PR TITLE
Fix renaming of non-renamble mounts

### PR DESCRIPTION
**Pull request by icewind1991**
- Properly block attempts to rename non movable mountpoint roots
- Fix the parsing of dav permissions so non-movable mounts dont show as movable in the web ui

Fixes https://github.com/nextcloud/groupfolders/issues/33

### CODE REVIEW

File: lib/private/Files/View.php [hunk 1]
```diff
@@ -779,14 +779,18 @@ public function rename($path1, $path2) {
...
-					if ($this->isTargetAllowed($absolutePath2)) {
-						/**
-						 * @var \OC\Files\Mount\MountPoint | \OC\Files\Mount\MoveableMount $mount1
-						 */
-						$sourceMountPoint = $mount1->getMountPoint();
-						$result = $mount1->moveMount($absolutePath2);
-						$manager->moveMount($sourceMountPoint, $mount1->getMountPoint());
+				if ($internalPath1 === '') {
+					if ($mount1 instanceof MoveableMount) {
+						if ($this->isTargetAllowed($absolutePath2)) {
```
Comment (nickvergessen) [inline lib/private/Files/View.php]: merge with the `mount1 instanceof MoveableMount` if above?

Comment (icewind1991) [inline lib/private/Files/View.php]: That would require a seperate `($internalPath1 === '' && !$mount1 instanceof MoveableMount)` case, imo this one is clearer than splitting it up

**Review by rullzer — APPROVED** [review]
Makes sense!

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

**Comment by yamlfd** [comment]
May this go into a bugfix release of 12.x or do we need to wait until 13.x?

**Comment by MorrisJobke** [comment]
> May this go into a bugfix release of 12.x or do we need to wait until 13.x?

It is already backported to stable12 and will land in 12.0.1: #5757

**Pull request merged**